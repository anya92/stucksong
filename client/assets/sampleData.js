export const recentlyPlayedTracks = [
  {
    album: 'Americana',
    artist: 'The Offspring',
    id: '4EchqUKQ3qAQuRNKmeIpnf',
    image: 'https://i.scdn.co/image/c77a1f3a2de72c2a8fde481a016376ca8ef81b33',
    played_at: '2018-08-14T18:50:09.757Z',
    title: 'The Kids Aren\'t Alright',
    uri: 'spotify:track:4EchqUKQ3qAQuRNKmeIpnf',
  },
  {
    album: 'No Pads, No Helmets...Just Balls (15th Anniversary Tour Edition)',
    artist: 'Simple Plan',
    id: '6otiaV2fagE3s8IvP6WkwG',
    image: 'https://i.scdn.co/image/9f6a521519080242f2de1c2b8873eec5372f7a6a',
    played_at: '2018-08-14T18:46:04.608Z',
    title: 'I\'m Just A Kid',
    uri: 'spotify:track:6otiaV2fagE3s8IvP6WkwG',
  },
  {
    album: 'All Killer, No Filler',
    artist: 'Sum 41',
    id: '1HNE2PX70ztbEl6MLxrpNL',
    image: 'https://i.scdn.co/image/4fc4f0bb924338f85e0fe7c899cde32ec5e9f10d',
    played_at: '2018-08-14T18:42:32.417Z',
    title: 'In Too Deep',
    uri: 'spotify:track:1HNE2PX70ztbEl6MLxrpNL',
  },
];

export const topArtists = [
  {
    genres: [
      'dance pop',
      'folk-pop',
      'indie pop',
      'indie r&b',
      'indietronica',
      'metropopolis',
      'pop',
    ],
    id: '4Ui2kfOqGujY81UcPrb5KE',
    image: 'https://i.scdn.co/image/a20acc6267aec9d08aa32d05ae2fb8647943f81a',
    name: 'HAIM',
    uri: 'spotify:artist:4Ui2kfOqGujY81UcPrb5KE',
  },
  {
    genres: [
      'chamber pop',
      'indie folk',
      'shimmer pop',
    ],
    id: '46CitWgnWrvF9t70C2p1Me',
    image: 'https://i.scdn.co/image/8b6e70aafad73059422920cbe528ecbcdff74018',
    name: 'Daughter',
    uri: 'spotify:artist:46CitWgnWrvF9t70C2p1Me',
  },
  {
    genres: [
      'alt-indie rock',
      'garage rock',
      'indie rock',
      'indietronica',
      'modern alternative rock',
      'modern rock',
    ],
    id: '3btzEQD6sugImIHPMRgkwV',
    image: 'https://i.scdn.co/image/f54c5e6b3919940e204406ef6846ee8e197f8217',
    name: 'Wolf Alice',
    uri: 'spotify:artist:3btzEQD6sugImIHPMRgkwV',
  },
];

export const topTracks = [
  {
    album: 'Gone Now',
    artist: 'Bleachers',
    id: '3ySU5vwQB33iGulwcUL9qQ',
    image: 'https://i.scdn.co/image/683a338f942640da90a27169e87c29440f20c4bb',
    title: 'Don\'t Take The Money',
    uri: 'spotify:track:3ySU5vwQB33iGulwcUL9qQ',
  },
  {
    album: 'Something To Tell You',
    artist: 'HAIM',
    id: '20QoFTI0fp2dNNexlHDhTe',
    image: 'https://i.scdn.co/image/9925a5f35b0a0e7b6f16e8f12c80cb5b4f015d57',
    title: 'Walking Away',
    uri: 'spotify:track:20QoFTI0fp2dNNexlHDhTe',
  },
  {
    album: 'Big Mess',
    artist: 'Grouplove',
    id: '1Tcd9flKIOxXWqR28aDP3U',
    image: 'https://i.scdn.co/image/781fa2da2c0dc8fc64f093dd753d4acfd6647db1',
    title: 'Don\'t Stop Making It Happen',
    uri: 'spotify:track:1Tcd9flKIOxXWqR28aDP3U',
  },
];

export const playlist = {
  name: 'Top Tracks',
  description: '',
  image: 'https://mosaic.scdn.co/640/129c6e4a622dc50c0802042780f92631a9dcc14056cc004f3ff47cff0ad8fcf05461730a193d8d8263b1ad734c51e06c7faee7fc95fd3abf0bbb9bbe6f4c08a3f6b73b062d6ea8384add3e3889814448',
  url: 'https://open.spotify.com/playlist/6yZesRjpu9OPKCM77N2EQt',
  tracks: topTracks,
};
