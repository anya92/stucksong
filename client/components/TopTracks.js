import React from 'react';
import { connect } from 'react-redux';
import {
  arrayOf,
  shape,
  string,
  func,
  bool,
} from 'prop-types';
import { fetchTracks } from '../actions';

import InfiniteScroll from './InfiniteScroll';
import Card from './Card';

import { CardsGrid } from '../styles/cards';
import Loader from '../styles/loader';

const TopTracks = ({
  topTracks: {
    tracks,
    hasMore,
    pending,
    error,
  },
  fetchTracks,
}) => (
  <CardsGrid>
    <InfiniteScroll
      loadMore={() => fetchTracks(tracks.length)}
      isLoading={pending}
      hasMore={hasMore}
    >
      <React.Fragment>
        { tracks.length > 0 && <h1>Your Top Tracks</h1> }
        {
          tracks.map((track, i) => (
            <Card key={`${track.id}${i}`} data={track} type="track" index={i} />
          ))
        }
        { pending && <Loader>Loading...</Loader> }
      </React.Fragment>
    </InfiniteScroll>
  </CardsGrid>
);


TopTracks.propTypes = {
  topTracks: shape({
    pending: bool.isRequired,
    error: bool.isRequired,
    hasMore: bool.isRequired,
    tracks: arrayOf(shape({
      id: string.isRequired,
      title: string.isRequired,
      artist: string.isRequired,
      album: string.isRequired,
      image: string.isRequired,
    })).isRequired,
  }).isRequired,
  fetchTracks: func.isRequired,
};

function mapStateToProps(state) {
  return {
    topTracks: state.topTracks,
  };
}

export default connect(mapStateToProps, { fetchTracks })(TopTracks);
