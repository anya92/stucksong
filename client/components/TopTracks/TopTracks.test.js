import React from 'react';
import { shallow } from 'enzyme';

import { TopTracks } from './TopTracks';
import { topTracks as tracks } from '../../assets/sampleData';

describe('TopTracks component', () => {
  let fetchTracks;
  beforeEach(() => {
    fetchTracks = jest.fn();
  });

  it('renders TopTracks with no data correctly', () => {
    const mockProps = {
      topTracks: {
        tracks: [],
        hasMore: true,
        pending: false,
        error: false,
      },
      fetchTracks,
    };
    const wrapper = shallow(<TopTracks {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders TopTracks with tracks correctly', () => {
    const mockProps = {
      topTracks: {
        tracks,
        hasMore: true,
        pending: false,
        error: false,
      },
      fetchTracks,
    };
    const wrapper = shallow(<TopTracks {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders TopTracks with pending correctly', () => {
    const mockProps = {
      topTracks: {
        tracks,
        hasMore: true,
        pending: true,
        error: false,
      },
      fetchTracks,
    };
    const wrapper = shallow(<TopTracks {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders TopTracks with pending correctly', () => {
    const mockProps = {
      topTracks: {
        tracks,
        hasMore: true,
        pending: false,
        error: 'Request failed',
      },
      fetchTracks,
    };
    const wrapper = shallow(<TopTracks {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
