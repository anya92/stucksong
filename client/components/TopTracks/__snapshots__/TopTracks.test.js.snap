// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTracks component renders TopTracks with no data correctly 1`] = `
<glamorous(div)>
  <NoData />
  <InfiniteScroll
    hasMore={true}
    isLoading={false}
    loadMore={[Function]}
  />
</glamorous(div)>
`;

exports[`TopTracks component renders TopTracks with pending correctly 1`] = `
<glamorous(div)>
  <InfiniteScroll
    hasMore={true}
    isLoading={true}
    loadMore={[Function]}
  >
    <h1>
      Your Top Tracks
    </h1>
    <Card
      data={
        Object {
          "album": "Gone Now",
          "artist": "Bleachers",
          "id": "3ySU5vwQB33iGulwcUL9qQ",
          "image": "https://i.scdn.co/image/683a338f942640da90a27169e87c29440f20c4bb",
          "title": "Don't Take The Money",
          "uri": "spotify:track:3ySU5vwQB33iGulwcUL9qQ",
        }
      }
      index={0}
      key="3ySU5vwQB33iGulwcUL9qQ0"
      type="track"
    />
    <Card
      data={
        Object {
          "album": "Something To Tell You",
          "artist": "HAIM",
          "id": "20QoFTI0fp2dNNexlHDhTe",
          "image": "https://i.scdn.co/image/9925a5f35b0a0e7b6f16e8f12c80cb5b4f015d57",
          "title": "Walking Away",
          "uri": "spotify:track:20QoFTI0fp2dNNexlHDhTe",
        }
      }
      index={1}
      key="20QoFTI0fp2dNNexlHDhTe1"
      type="track"
    />
    <Card
      data={
        Object {
          "album": "Big Mess",
          "artist": "Grouplove",
          "id": "1Tcd9flKIOxXWqR28aDP3U",
          "image": "https://i.scdn.co/image/781fa2da2c0dc8fc64f093dd753d4acfd6647db1",
          "title": "Don't Stop Making It Happen",
          "uri": "spotify:track:1Tcd9flKIOxXWqR28aDP3U",
        }
      }
      index={2}
      key="1Tcd9flKIOxXWqR28aDP3U2"
      type="track"
    />
    <glamorous(div)>
      Loading...
    </glamorous(div)>
  </InfiniteScroll>
</glamorous(div)>
`;

exports[`TopTracks component renders TopTracks with pending correctly 2`] = `
<LoadableComponent
  error="Request failed"
/>
`;

exports[`TopTracks component renders TopTracks with tracks correctly 1`] = `
<glamorous(div)>
  <InfiniteScroll
    hasMore={true}
    isLoading={false}
    loadMore={[Function]}
  >
    <h1>
      Your Top Tracks
    </h1>
    <Card
      data={
        Object {
          "album": "Gone Now",
          "artist": "Bleachers",
          "id": "3ySU5vwQB33iGulwcUL9qQ",
          "image": "https://i.scdn.co/image/683a338f942640da90a27169e87c29440f20c4bb",
          "title": "Don't Take The Money",
          "uri": "spotify:track:3ySU5vwQB33iGulwcUL9qQ",
        }
      }
      index={0}
      key="3ySU5vwQB33iGulwcUL9qQ0"
      type="track"
    />
    <Card
      data={
        Object {
          "album": "Something To Tell You",
          "artist": "HAIM",
          "id": "20QoFTI0fp2dNNexlHDhTe",
          "image": "https://i.scdn.co/image/9925a5f35b0a0e7b6f16e8f12c80cb5b4f015d57",
          "title": "Walking Away",
          "uri": "spotify:track:20QoFTI0fp2dNNexlHDhTe",
        }
      }
      index={1}
      key="20QoFTI0fp2dNNexlHDhTe1"
      type="track"
    />
    <Card
      data={
        Object {
          "album": "Big Mess",
          "artist": "Grouplove",
          "id": "1Tcd9flKIOxXWqR28aDP3U",
          "image": "https://i.scdn.co/image/781fa2da2c0dc8fc64f093dd753d4acfd6647db1",
          "title": "Don't Stop Making It Happen",
          "uri": "spotify:track:1Tcd9flKIOxXWqR28aDP3U",
        }
      }
      index={2}
      key="1Tcd9flKIOxXWqR28aDP3U2"
      type="track"
    />
  </InfiniteScroll>
</glamorous(div)>
`;
